services/property-svc/internal/property/aggregate/aggregate.go

```diff
@@ -182,15 +184,56 @@ func (a *PropertyAggregate) onFieldTypeDataSelectOptionsUpserted(evt hwes.Event)
 		return err
 	}
 
-	if a.Property.FieldTypeData.SelectData != nil {
-		// TODO: Check and update already existing SelectOptions don't just append them
-		a.Property.FieldTypeData.SelectData.SelectOptions = append(a.Property.FieldTypeData.SelectData.SelectOptions, payload.UpsertedSelectOptions...)
-	} else {
-		a.Property.FieldTypeData.SelectData = &models.SelectData{
-			SelectOptions: payload.UpsertedSelectOptions,
+	var initialAllowFreetext bool
+	existingSelectOptions := make(map[uuid.UUID]models.SelectOption)
+
+	initialFieldTypeData := a.Property.FieldTypeData
+	if initialFieldTypeData != nil {
+		if initialFieldTypeData.SelectData != nil {
+			initialAllowFreetext = initialFieldTypeData.SelectData.AllowFreetext
+			for _, option := range initialFieldTypeData.SelectData.SelectOptions {
+				existingSelectOptions[option.ID] = option
+			}
+		}
+	}
+
+	for _, option := range payload.UpsertedSelectOptions {
+		if _, exists := existingSelectOptions[option.ID]; !exists {
+			if option.Name == nil {
+				return fmt.Errorf("name missing for selectOption with id %s", option.ID.String())
+			}
+			existingSelectOptions[option.ID] = models.SelectOption{
+				ID:          option.ID,
+				Name:        *option.Name,
+				Description: option.Description,
+			}
+		} else {
+			updatedOpt := existingSelectOptions[option.ID]
+			if option.Name != nil {
+				updatedOpt.Name = *option.Name
+			}
+			if option.Description != nil {
+				updatedOpt.Description = option.Description
+			}
+			if option.IsCustom != nil {
+				updatedOpt.IsCustom = *option.IsCustom
+			}
+			existingSelectOptions[option.ID] = updatedOpt
 		}
 	}
 
+	var updatedSelectOptions []models.SelectOption
+	for _, opt := range existingSelectOptions {
+		updatedSelectOptions = append(updatedSelectOptions, opt)
+	}
+
+	a.Property.FieldTypeData = &models.FieldTypeData{
```

Comment:
can we, instead of creating a new FieldTypeData struct, update the existing one?
Say we add another value to the struct later on, then it would be removed here.